[Offload] Correctly reject building on unsupported architectures (#92276)
llvm/llvm-project: offload/CMakeLists.txt

  project(offload C CXX ASM)
endif()

# Check that the library can actually be built.
if(APPLE OR WIN32 OR WASM)
  message(WARNING "libomptarget cannot be built on Windows and MacOS X!")
  return()
elseif(NOT "cxx_std_17" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
  message(WARNING "Host compiler must support C++17 to build libomptarget!")
  return()
elseif(NOT CMAKE_SIZEOF_VOID_P EQUAL 8)

>   message(WARNING "libomptarget on 32-bit systems are not yet supported!")

Reviewer: yet ? 
are you planning to implement it?
if not, i would just remove the yet
```suggestion
  message(WARNING "libomptarget on 32-bit systems is not supported!")
```

Author: It's theoretically possible to implement, since maybe some day someone will have a use-case for it. But since there's no active plans for it I suppose it's best to keep it simple.
